lucene/core/src/java/org/apache/lucene/index/MergePolicy.java

```diff
@@ -399,8 +423,19 @@ public String segString(Directory dir) {
       }
       return b.toString();
     }
+
+    boolean await(long timeout, TimeUnit unit) {
+      for (OneMerge merge : merges) {
+        if (merge.await(timeout, unit) == false) {
```

Comment:
This looks suspicious when there is more than one merge.  Shouldn't the timeout decrease as time is used up by earlier merges?

In practice, when is there more than one?  I've been confused on this matter when I developed a custom MP/MS.

Reply:
in a real change that's correct. in a prototype as this is it's really just there to visualize the idea. I didn't do this on purpose to not discuss impl details. that's not the point of this it's really just a PR to make commenting simpler.

Reply:
I fixed this in the followup PR https://github.com/apache/lucene-solr/pull/1585